integration/simulation/transaction_injector.go

```diff
@@ -245,38 +233,55 @@ func (m *TransactionInjector) issueInvalidWithdrawals() {
 	}
 }
 
-// Uses one of three approaches to create an invalidly-signed transaction.
-func createInvalidSignature(tx *nodecommon.L2Tx, fromWallet *wallet_mock.Wallet) *nodecommon.L2Tx {
```

Comment:
did the logic in this function move somewhere?

Reply:
Removed the 1st and set a TODO for the 3rd case. Left the 2nd case on.

Reply:
I don't understand what was wrong with it. Did it break anything?

Reply:
It would break the wallet package has it doesn't expose the PrivateKey.
Also, case 0 didn't make much sense to repeatedly test. Independently of the payload if a signature is made with a different ChainID this will be an ignored withdrawal.
And Case 2 is not really applicable because the withdrawal is only looking at the payload not at who signed it.

Reply:
I exposed the private key and made the case 0 available.

I'm not sure case 2 is actually doing anything relevant, looking more into it.

Reply:
Removed case 2. After a bit of digging wrote a unit test to see what was happening and It's returning an unsigned tx same as case 1 👍 

I don't think it's worth adding the unit test in, but let me know if you think we should add it in.
Pasting it here for reference.

```gopackage simulation

import (
	"fmt"
	"math/big"
	"testing"

	"github.com/ethereum/go-ethereum/core/types"

	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
	"github.com/obscuronet/obscuro-playground/go/obscuronode/enclave/core"
	wallet_mock "github.com/obscuronet/obscuro-playground/integration/walletmock"
)

func TestNewOutputStats(t *testing.T) {
	fromWallet := wallet_mock.New()

	txData := core.L2TxData{Type: core.WithdrawalTx, From: fromWallet.Address, Amount: obscurocommon.RndBtw(1, 100)}
	newTx := wallet_mock.NewL2Tx(txData)
	signedTx := wallet_mock.SignTx(newTx, fromWallet.Key.PrivateKey)
	// After signing the transaction, we create a new transaction based on the transaction data, breaking the signature.
	retTx := wallet_mock.NewL2Tx(txData)

	signedV, signedR, signedS := signedTx.RawSignatureValues()
	fmt.Printf("SignedTx: %v %v %v  \n", signedV, signedR, signedS)
	if signedV.Int64() == 0 || signedR.Int64() == 0 || signedS.Int64() == 0 {
		t.Fatal("unexpected values")
	}

	err := verifySignature(signedTx)
	if err != nil {
		t.Fatal(err)
	}

	retTxV, retTxR, retTxS := retTx.RawSignatureValues()
	fmt.Printf("RetTx: %v %v %v  \n", retTxV, retTxR, retTxS)
	if retTxV.Int64() != 0 || retTxR.Int64() != 0 || retTxS.Int64() != 0 {
		t.Fatal("unexpected values")
	}

	err = verifySignature(retTx)
	if err == nil {
		t.Fatal(err)
	}
}

// Checks that the L2Tx has a valid signature.
func verifySignature(decryptedTx *types.Transaction) error {
	signer := types.NewLondonSigner(big.NewInt(777))
	_, err := types.Sender(signer, decryptedTx)
	return err
}

```